### ISSUE TITLE
Another display

### ISSUE THREAD

**Issue by kpbicka** [djspawnbrest/ESP32AY#1]
Добрый день. Возможно ли пересобрать под использование дисплеев на контроллере ILI9341 ?

**Comment by djspawnbrest** [comment]
> Добрый день. Возможно ли пересобрать под использование дисплеев на контроллере ILI9341 ?

Да, библиотека поддерживает этот контроллер, надо только в библиотеке его выбрать и прописать настройки. Соответственно пины проверить на совместимость.

**Comment by kpbicka** [comment]
> > Добрый день. Возможно ли пересобрать под использование дисплеев на контроллере ILI9341 ?
> 
> Да, библиотека поддерживает этот контроллер, надо только в библиотеке его выбрать и прописать настройки. Соответственно пины проверить на совместимость.

А ещё расскажите, пожалуйста, как оно работает по UART (ну или где это описано - не нашёл)

**Comment by kpbicka** [comment]
И в веб-прошивальщике неплохо бы сделать выбор скорости, а то не всегда получается на той, что по умолчанию (которая не отображается) :)

**Comment by djspawnbrest** [comment]
> > > Добрый день. Возможно ли пересобрать под использование дисплеев на контроллере ILI9341 ?
> > 
> > 
> > Да, библиотека поддерживает этот контроллер, надо только в библиотеке его выбрать и прописать настройки. Соответственно пины проверить на совместимость.
> 
> А ещё расскажите, пожалуйста, как оно работает по UART (ну или где это описано - не нашёл)

Заходим в настройки (двойной клик энкодера из плеера), там верхний пункт меню "Player Source" - переключает источник проигрывания SD/UART. Качаем плеер AVR-AY Player, устанвливаем, запускаем его, снимаем галочку "USE SOUND CARD", выбираем ком порт подключенного плеера - Open. Запускаем треки из плейлиста AVR-AY Player'a.....

**Comment by djspawnbrest** [comment]
> И в веб-прошивальщике неплохо бы сделать выбор скорости, а то не всегда получается на той, что по умолчанию (которая не отображается) :)

Такой реализации у разработчика пока нет. Скорее всего у Вас уже уставший провод.

**Comment by kpbicka** [comment]
> 

а почему у вас в User_Setup.h
#define ILI9341_DRIVER  
но используется вроде ST7789 ?

**Comment by djspawnbrest** [comment]
> > 
> 
> а почему у вас в User_Setup.h #define ILI9341_DRIVER но используется вроде ST7789 ?

Потому что это всего лишь название файла с его id, а если посмотреть его внутри то там вообще прописан [#define ST7789_DRIVER](https://github.com/djspawnbrest/ESP32AY/blob/main/lib/TFT_eSPI/User_Setups/Setup42_ILI9341_ESP32.h#L7). Просто по набору настроек больше подходил мне, вот его я и переписал. И user_setup.h не используется, используется user_setup_select

**Comment by mszulen** [comment]
And what, exactly display did you use? Any chance for a link to the seller? Is this the correct one? 2.4 inch ST7789-no touch with 18 pin ribbon https://www.aliexpress.com/item/1005005824438283.html

**Comment by djspawnbrest** [comment]
> And what, exactly display did you use? Any chance for a link to the seller?

For example use this: https://aliexpress.ru/item/1005005824438283.html?sku_id=12000034482807420&spm=a2g2w.productlist.search_results.0.318d3d31onp0Zm
or
https://aliexpress.ru/item/1005005287329824.html?sku_id=12000032498295116&spm=a2g2w.productlist.search_results.8.318d3d31onp0Zm

It doesn't matter with touch or without, it is not used.

**Comment by mszulen** [comment]
Perfect! Thank you! :)
